SingleNumber.js

```diff
@@ -0,0 +1,15 @@
+
+   function findSingle(ar, size)
+    {
+
+        let res = ar[0];
+    for (let i = 1; i < size; i++)
+    res = res ^ ar[i];
+
+    return res;
+	}
+
+    let ar = [4, 4, 5, 5, 7, 3, 3];
+    let n = ar.length;
```

Comment:
We could take the array length within `findSingle` function itself to avoid mismatch if we pass a wrong size.